test/e2e/operator/deployment_api.go

```diff
@@ -96,16 +98,22 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 		evWatcher, err = f.ClientSet.CoreV1().Events("").Watch(context.TODO(), metav1.ListOptions{})
 		Expect(err).ShouldNot(HaveOccurred(), "get events watcher")
 		evCaptured = map[types.UID]map[string]struct{}{}
+		evMutex = sync.Mutex{}
 		go func() {
 			for watchEvent := range evWatcher.ResultChan() {
 				ev, ok := watchEvent.Object.(*corev1.Event)
 				if !ok || ev.Source.Component != "pmem-csi-operator" {
 					continue
 				}
-				if _, ok := evCaptured[ev.InvolvedObject.UID]; !ok {
-					evCaptured[ev.InvolvedObject.UID] = map[string]struct{}{}
-				}
-				evCaptured[ev.InvolvedObject.UID][ev.Reason] = struct{}{}
+				go func() {
```

Comment:
Why a goroutine here?

If you want to lock the mutex and unlock when done, then just wrap the map access with Lock/Unlock:
```
   evMutex.Lock()
   if _, ok := evCaptured[ev.InvolvedObject.UID]; !ok {
   ...
   }
   evMutex.Unlock()
```

But I doubt that this is actually solving anything, see my other comment.

    evMutex.Lock()

Reply:
`defer` is only useful when there are multiple return points. This map usage is so simple that we don't need it.